### PR TITLE
Review request for Microsoft.Insights to add package 2022-08-23

### PR DESCRIPTION
**Pull request by tokaplan**
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by tokaplan** [comment]
@akning-ms @raosuhas I fixed all the issues I could. The checks that fail - I don't really understand them, and not sure if they are internal tooling issues or relevant problems. Can you please take a look? I've organized the PR so that you can easily review the actual changes.

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json [hunk 1]
```diff
@@ -0,0 +1,1199 @@
...
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Insights/privateLinkScopes": {
+      "get": {
+        "description": "Gets a list of all Azure Monitor PrivateLinkScopes within a subscription.",
+        "operationId": "PrivateLinkScopes_List",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
```
Comment (rkmanda) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json:43]: >v2 [](http://example.com/codeflow?start=69&length=2)

use v3 types everywhere

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json [hunk 2]
```diff
@@ -0,0 +1,1199 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "ApplicationInsightsManagementClient",
+    "description": "Azure Application Insights API reference for Private Links Scopes management.",
+    "version": "2021-09-01"
```
Comment (rkmanda) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json:6]: >2021-09-01 [](http://example.com/codeflow?start=16&length=10)

The title suggests 2022-08-23 is being introduced whereas the version number here is 2021-09-01. Which of these is the correct one?

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json [hunk 3]
```diff
@@ -0,0 +1,1199 @@
...
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Insights/privateLinkScopes": {
+      "get": {
+        "description": "Gets a list of all Azure Monitor PrivateLinkScopes within a subscription.",
+        "operationId": "PrivateLinkScopes_List",
```
Comment (rkmanda) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json:40]: >PrivateLinkScopes_List [](http://example.com/codeflow?start=24&length=22)

PrivateLinkScopes_ListBySubscription

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json [hunk 4]
```diff
@@ -0,0 +1,1199 @@
...
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Insights/privateLinkScopes": {
+      "get": {
+        "description": "Gets a list of Azure Monitor PrivateLinkScopes within a resource group.",
+        "operationId": "PrivateLinkScopes_ListByResourceGroup",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ResourceGroupNameParameter"
```
Comment (rkmanda) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json:79]: >v2 [](http://example.com/codeflow?start=69&length=2)

use v3

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json [hunk 5]
```diff
@@ -0,0 +1,1199 @@
...
+      "delete": {
+        "description": "Deletes a Azure Monitor PrivateLinkScope.",
+        "operationId": "PrivateLinkScopes_Delete",
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "azure-async-operation"
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ResourceGroupNameParameter"
```
Comment (rkmanda) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json:122]: >v2 [](http://example.com/codeflow?start=69&length=2)

v3

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json [hunk 6]
```diff
@@ -0,0 +1,1199 @@
...
+            "$ref": "./examples/PrivateLinkScopesDelete.json"
+          }
+        }
+      },
+      "get": {
+        "description": "Returns a Azure Monitor PrivateLinkScope.",
+        "operationId": "PrivateLinkScopes_Get",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ResourceGroupNameParameter"
```
Comment (rkmanda) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json:167]: >v2 [](http://example.com/codeflow?start=69&length=2)

v3 - change everywhere

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json [hunk 7]
```diff
@@ -0,0 +1,1199 @@
...
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful request when creating or updating a Azure Monitor PrivateLinkScope. The updated PrivateLinkScope is returned.",
+            "schema": {
+              "$ref": "#/definitions/AzureMonitorPrivateLinkScope"
+            }
+          },
+          "201": {
+            "description": "Successful request when creating or updating a Azure Monitor PrivateLinkScope. The updated PrivateLinkScope was created and is returned.",
```
Comment (rkmanda) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json:233]: >or updating [](http://example.com/codeflow?start=61&length=11)

remove "or updating". 201 should be returned only on create and not on update

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json [hunk 8]
```diff
@@ -0,0 +1,1199 @@
...
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/AzureMonitorPrivateLinkScope"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful request when creating or updating a Azure Monitor PrivateLinkScope. The updated PrivateLinkScope is returned.",
```
Comment (rkmanda) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json:227]: >creating or [](http://example.com/codeflow?start=52&length=11)

remove "creating or". 200 should be returned only on update and not create

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json [hunk 9]
```diff
@@ -0,0 +1,1199 @@
...
+          "PrivateLinkScopeCreate": {
+            "$ref": "./examples/PrivateLinkScopesCreate.json"
+          },
+          "PrivateLinkScopeUpdate": {
+            "$ref": "./examples/PrivateLinkScopesUpdate.json"
+          }
+        }
+      },
+      "patch": {
+        "description": "Updates an existing PrivateLinkScope's tags. To update other fields use the CreateOrUpdate method.",
```
Comment (rkmanda) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json:255]: >update [](http://example.com/codeflow?start=72&length=6)

replace all fields

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json [hunk 10]
```diff
@@ -0,0 +1,1199 @@
...
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully approved or rejected private endpoint connection.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/privatelinks.json#/definitions/PrivateEndpointConnection"
+            }
+          },
+          "202": {
```
Comment (rkmanda) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json:528]: >202 [](http://example.com/codeflow?start=11&length=3)

201 is the recommended status code for long running PUTs. Is the NRP contract forcing you to make this a 202 ?

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json [hunk 11]
```diff
@@ -0,0 +1,1199 @@
...
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/PrivateLinkScopeName"
+          }
+        ],
+        "responses": {
```
Comment (rkmanda) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json:630]: >responses [](http://example.com/codeflow?start=9&length=9)

include x-ms-pageable for all list results

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json [hunk 12]
```diff
@@ -0,0 +1,1199 @@
...
+            }
+          },
+          "201": {
+            "description": "Successfully created a new azure monitor scoped resource in a private link scope.",
+            "schema": {
+              "$ref": "#/definitions/ScopedResource"
+            }
+          },
+          "202": {
+            "description": "Accepted",
```
Comment (rkmanda) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json:752]: long running PUTs MUST only return a 201 and not a 202. Remove 202.

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json [hunk 13]
```diff
@@ -0,0 +1,1199 @@
...
+          "x-nullable": true,
+          "description": "Start time of the job in standard ISO8601 format."
+        },
+        "endTime": {
+          "type": "string",
+          "format": "date-time",
+          "x-nullable": true,
+          "description": "End time of the job in standard ISO8601 format."
+        },
+        "status": {
```
Comment (rkmanda) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json:903]: >status [](http://example.com/codeflow?start=9&length=6)

This needs to be an enum with the 3 min values defined as described in this link [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#azure-asyncoperation-resource-format](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#azure-asyncoperation-resource-format)

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json [hunk 14]
```diff
@@ -0,0 +1,1199 @@
...
+        },
+        "systemData": {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData",
+          "description": "System data",
+          "readOnly": true
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/TrackedResource"
```
Comment (rkmanda) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json:942]: >v2 [](http://example.com/codeflow?start=67&length=2)

switch to using v3 type

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json [hunk 15]
```diff
@@ -0,0 +1,1199 @@
...
+      "description": "A container holding only the Tags for a resource, allowing the user to update the tags on a PrivateLinkScope instance."
+    },
+    "AzureMonitorPrivateLinkScope": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Properties that define a Azure Monitor PrivateLinkScope resource.",
+          "$ref": "#/definitions/AzureMonitorPrivateLinkScopeProperties"
+        },
+        "systemData": {
```
Comment (rkmanda) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json:934]: >systemData [](http://example.com/codeflow?start=9&length=10)

drop systemData from the properties bag as it is already defined in the TrackedResource common type definition

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json [hunk 16]
```diff
@@ -0,0 +1,1199 @@
...
+          },
+          "description": "Current state of this PrivateLinkScope: whether or not is has been provisioned within the resource group it is defined. Users cannot change this value but are able to read from it. Values will include Provisioning ,Succeeded, Canceled and Failed.",
+          "readOnly": true
+        },
+        "privateEndpointConnections": {
+          "readOnly": true,
+          "description": "List of private endpoint connections.",
+          "type": "array",
+          "items": {
+            "$ref": "../../../../../common-types/resource-management/v2/privatelinks.json#/definitions/PrivateEndpointConnection"
```
Comment (rkmanda) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json:994]: >PrivateEndpointConnection [](http://example.com/codeflow?start=103&length=25)

This is defined as a separate proxy resource. Is the NRP contract forcing this to be present inside the properties bag of the privateLinkScope resource type?

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json [hunk 17]
```diff
@@ -0,0 +1,1199 @@
...
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/ScopedResourceProperties",
+          "description": "Resource properties.",
+          "x-ms-client-flatten": true
+        },
+        "systemData": {
```
Comment (rkmanda) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json:1116]: >systemData [](http://example.com/codeflow?start=9&length=10)

remove this after changing ProxyResource to point to v3 type

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json [hunk 18]
```diff
@@ -0,0 +1,1199 @@
...
+                "description": "The state is Provisioning.",
+                "name": "Provisioning"
+              }
+            ]
+          },
+          "readOnly": true
+        }
+      }
+    },
+    "DefaultErrorResponse": {
```
Comment (rkmanda) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json:1169]: >DefaultErrorResponse [](http://example.com/codeflow?start=5&length=20)

replace with "ErrorResponse" common type

File: specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json [hunk 19]
```diff
@@ -0,0 +1,1199 @@
...
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/PrivateLinkScopeName"
+          }
+        ],
+        "responses": {
+          "200": {
```
Comment (rkmanda) [inline specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2021-09-01/privateLinkScopes_API.json:369]: >200 [](http://example.com/codeflow?start=11&length=3)

add x-ms-pageable

**Comment by akning-ms** [comment]
@jianyexi to help take the exception in Linter validation

**Pull request closed without merging**